service-base/src/main/java/org/eclipse/hono/service/auth/device/DeviceCertificateValidator.java

```diff
@@ -45,14 +47,40 @@
      */
     @Override
     public Future<Void> validate(final List<X509Certificate> chain, final TrustAnchor trustAnchor) {
+        return validate(chain, List.of(trustAnchor));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Future<Void> validate(final List<X509Certificate> chain, final List<TrustAnchor> trustAnchors) {
 
         Objects.requireNonNull(chain);
-        Objects.requireNonNull(trustAnchor);
+        Objects.requireNonNull(trustAnchors);
 
-        if (chain.isEmpty()) {
-            throw new IllegalArgumentException("certificate chain must not be empty");
+        if (chain.isEmpty() || trustAnchors.isEmpty()) {
+            throw new IllegalArgumentException("certificate chain and/or trust anchors must not be empty");
         }
 
+        final Future<Void> result = Future.future();
+        final List<Future<Void>> futures = new ArrayList<>();
+
+        trustAnchors.forEach(trustAnchor -> {
+            futures.add(doValidate(chain, trustAnchor));
+        });
+        CompositeFutureImpl.any(futures.toArray(Future[]::new)).setHandler(validationCheck -> {
+            if (validationCheck.succeeded()) {
+                result.complete();
+            } else {
+                result.fail(validationCheck.cause());
+            }
+        });
+        return result;
+    }
+
+    private Future<Void> doValidate(final List<X509Certificate> chain, final TrustAnchor trustAnchor) {
```

Comment:
FMPOV there is no need to iteratively invoke this method for each trust anchor.
Instead, simply pass in all trust anchors to the constructor for the `PKIXParameters` ...

Reply:
@sophokles73 if this looks good, i will move on to adding the ITs for the tenant service

Reply:
Yes, this looks good :+1: 
On to the next step ...